drivers/pwm/pwm_nrfx.c

```diff
@@ -121,6 +121,72 @@ static int pwm_nrfx_init(struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+
+static int pwm_nrfx_set_power_state(u32_t new_state, struct device *dev)
+{
+	if (new_state == DEVICE_PM_ACTIVE_STATE) {
+		int res = pwm_nrfx_init(dev);
+
+		assert(res == 0);
+	} else {
+		assert(new_state == DEVICE_PM_LOW_POWER_STATE ||
+		       new_state == DEVICE_PM_SUSPEND_STATE ||
+		       new_state == DEVICE_PM_OFF_STATE);
+		const struct pwm_nrfx_config *pconfig =
+						dev->config->config_info;
+
+		nrfx_pwm_uninit(&pconfig->pwm);
+	}
+
+	return 0;
+}
+
+static int pwm_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
+				void *context, u32_t *current_state)
+{
+	int err = 0;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t new_state = *((const u32_t *)context);
+
+		if (new_state != (*current_state)) {
+			err = pwm_nrfx_set_power_state(new_state, dev);
+			(*current_state) = new_state;
+		}
+	} else {
+		assert(ctrl_command == DEVICE_PM_GET_POWER_STATE);
+		*((u32_t *)context) = (*current_state);
+	}
+
+	return err;
+}
+
+#define PWM_NRFX_PM_CONTROL(idx)					\
+static int pwm_##idx##_nrfx_pm_control(struct device *dev,		\
+				       u32_t ctrl_command,		\
+				       void *context)			\
+{									\
+	static u32_t current_state = DEVICE_PM_ACTIVE_STATE;		\
+	return pwm_nrfx_pm_control(dev, ctrl_command, context,		\
+				   &current_state);			\
+}
+
+
+#ifdef CONFIG_PWM_0
+PWM_NRFX_PM_CONTROL(0)
```

Comment:
Instead of all these, just add `PWM_NRFX_PM_CONTROL(idx)` to the definition of `PWM_NRFX_DEVICE(idx)`.
And here:
```
#else

#define PWM_NRFX_PM_CONTROL(idx)

#endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
```

Reply:
done